regexp.go

```diff
@@ -437,3 +440,22 @@ func matchRules(text []rune, pos int, rules []*CompiledRule) (int, *CompiledRule
 	}
 	return 0, &CompiledRule{}, nil
 }
+
+// replace \r and \r\n with \n
+// same as strings.ReplaceAll but more efficient
+func ensureLF(text string) string {
+	buf := make([]byte, len(text))
+	var j int
+	for i := 0; i < len(text); i++ {
```

Comment:
I don't think this is unicode safe is it? If a unicode code point contains `\r\n` this will mangle the codepoint.

Reply:
Correct, this would iterate over the bytes (uint8s). We have to use a for range loop here, so we iterate over the individual runes.